### PR TITLE
[nrf toup] [nrfconnect] Add support for SUIT over Matter OTA

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Added support for multi-image dfu target while using SUIT.

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

File: config/nrfconnect/chip-module/Kconfig.defaults [hunk 1]
```diff
@@ -219,11 +219,11 @@ config BOOT_IMAGE_ACCESS_HOOKS
 
 config UPDATEABLE_IMAGE_NUMBER
     default 3 if NRF_WIFI_PATCHES_EXT_FLASH_STORE
-    default 2 if SOC_SERIES_NRF53X
-    default 1 if SUIT # SUIT does not support the multi-image dfu
+    default 2 if SOC_SERIES_NRF53X || CHIP_BOOTLOADER_SUIT
 
 config DFU_MULTI_IMAGE_MAX_IMAGE_COUNT
     default 3 if NRF_WIFI_PATCHES_EXT_FLASH_STORE
+    default 2 if CHIP_BOOTLOADER_SUIT
```
[the remark this answers is not in the document]
Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig.defaults:226]: This is done to ensure that SUIT will use only 2 images.

File: config/nrfconnect/chip-module/Kconfig [hunk 2]
```diff
@@ -104,10 +104,12 @@ config CHIP_DFU_LIBRARY_MCUMGR
 
 config CHIP_DFU_LIBRARY_DFU_TARGET
 	bool "Use dfu target library for Matter DFU purposes"
-	# MCUBOOT
-	imply DFU_MULTI_IMAGE if CHIP_BOOTLOADER_MCUBOOT
+	imply DFU_MULTI_IMAGE
```
[the remark this answers is not in the document]
Comment (ArekBalysNordic) [reply config/nrfconnect/chip-module/Kconfig:107]: Yes, we use it in 52 as well.

File: src/platform/nrfconnect/OTAImageProcessorImpl.cpp [hunk 3]
```diff
@@ -184,7 +181,11 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
             [](System::Layer *, void * /* context */) {
                 PlatformMgr().HandleServerShuttingDown();
                 k_msleep(CHIP_DEVICE_CONFIG_SERVER_SHUTDOWN_ACTIONS_SLEEP_MS);
+#ifdef CONFIG_DFU_TARGET_SUIT
+                dfu_target_suit_reboot();
```
Comment (Damian-Nordic) [inline src/platform/nrfconnect/OTAImageProcessorImpl.cpp:185]: The purpose of `Reboot` was also to save the reboot reason to "software update". Is this somehow handled for SUIT?

**Review by Damian-Nordic — APPROVED** [review]
Nice unification!

File: src/platform/nrfconnect/OTAImageProcessorImpl.cpp (hunk 3 above)
Comment (ArekBalysNordic) [reply src/platform/nrfconnect/OTAImageProcessorImpl.cpp:185]: No, it is not, we need to think how to handle it. I'm going to do it in the next iteration after the release. For now, it is base support.

**Pull request merged**